packages/compiler/src/core/checker.ts

```diff
@@ -924,6 +929,26 @@ export function createChecker(program: Program, resolver: NameResolver): Checker
     node: Node,
     mapper?: TypeMapper,
     valueConstraint?: CheckValueConstraint | undefined,
+  ): Type | Value | IndeterminateEntity | null {
+    const ret = checkNodeWorker(node, mapper, valueConstraint);
```

Comment:
why is this not just done in the checkTemplateParameter. 
This will just work for everything instead of making model a special case